packages-content-model/roosterjs-content-model-api/lib/modelApi/selection/adjustTrailingSpaceSelection.ts

```diff
@@ -0,0 +1,63 @@
+import { createText } from 'roosterjs-content-model-dom';
+import { iterateSelections } from 'roosterjs-content-model-core';
+import type {
+    ContentModelDocument,
+    ContentModelSegment,
+    ContentModelText,
+} from 'roosterjs-content-model-types';
+
+/**
+ * @internal
+ */
+export function adjustTrailingSpaceSelection(model: ContentModelDocument) {
+    iterateSelections(model, (_, __, block) => {
+        if (block?.blockType == 'Paragraph' && block.segments.length == 1) {
```

Comment:
If the selected paragraph has multiple segments, we don't handle it?

Reply:
The bug usually happens when the user does double click in a segment with trailing spaces. I made a change to cover more scenarios